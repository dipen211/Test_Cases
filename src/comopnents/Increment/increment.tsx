import React, { useState, useEffect } from 'react';

export default function Increment() {
    const [count, setCount] = useState(0);

    return (
        <div>
            <p>Count: {count}</p>
            <button
                onClick={() => {
                    setCount(count + 1);
                }}
            >
                Increment Count
            </button>
        </div>
    );
}