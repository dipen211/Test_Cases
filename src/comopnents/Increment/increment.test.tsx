import React from 'react';

import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Increment from './increment';
import { mountToJson } from 'enzyme-to-json';


Enzyme.configure({ adapter: new Adapter() });
describe('MyComponent', () => {
    const onCountChange = jest.fn();
    let wrapper: any;
    beforeEach(() => {
        wrapper = shallow(<Increment />);
    });
    it('should match the snapshot', () => {
        expect(mountToJson(wrapper)).toMatchSnapshot();
    });

    it('renders', () => {
        expect(wrapper).not.toBeNull();
    });

    it('shows my default text', () => {
        expect(wrapper.find('p').text()).toEqual('Count: 0');
    });

    it('correctly increments the count by 1', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('p').text()).toEqual('Count: 1');
    });

    it('correctly increments the count by 4', () => {
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');
        expect(wrapper.find('p').text()).toEqual('Count: 4');
    });
});