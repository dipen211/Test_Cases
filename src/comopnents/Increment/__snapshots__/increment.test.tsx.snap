// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should match the snapshot 1`] = `
<div>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment Count
  </button>
</div>
`;
